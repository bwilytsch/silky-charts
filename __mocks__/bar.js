/* ignore file coverage */

export const data = [
  { name: 'foo', value: 19 },
  { name: 'bar', value: 70 },
  { name: 'baz', value: 93 },
  { name: 'quz', value: 80 },
];
export const dataWidthDates = [
  { name: '2018-01-31T05:00:00.000Z', value: 19 },
  { name: '2018-02-01T05:00:00.000Z', value: 70 },
  { name: '2018-02-02T05:00:00.000Z', value: 93 },
  { name: '2018-02-03T05:00:00.000Z', value: 80 },
];
