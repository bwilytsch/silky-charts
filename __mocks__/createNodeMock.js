export default element => {
  if (element.type === 'svg') {
    return {
      parentElement: {},
    };
  }
};
