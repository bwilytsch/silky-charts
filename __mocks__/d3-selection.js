/* ignore file coverage */

const selection = jest.genMockFromModule('d3-selection');

const attr = () => ({ attr });

const fn = () => ({
  call: jest.fn(),
  select: jest.fn(() => ({ attr })),
  selectAll: jest.fn(() => ({ attr })),
});

selection.select = fn;
selection.selectAll = fn;

module.exports = selection;
