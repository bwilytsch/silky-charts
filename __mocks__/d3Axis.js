/* ignore file coverage */

module.exports = jest.fn(() => ({
  ticks: jest.fn(() => ({
    tickFormat: jest.fn(),
  })),
}));
