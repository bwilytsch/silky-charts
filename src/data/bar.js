export const letters = [
  { name: 'E', value: 0.12702 },
  { name: 'T', value: 0.09056 },
  { name: 'A', value: 0.08167 },
  { name: 'O', value: 0.07507 },
  { name: 'I', value: 0.06966 },
  { name: 'N', value: 0.06749 },
  { name: 'S', value: 0.06327 },
  { name: 'H', value: 0.06094 },
  { name: 'R', value: 0.05987 },
  { name: 'D', value: 0.04253 },
  { name: 'L', value: 0.04025 },
  { name: 'C', value: 0.02782 },
  { name: 'U', value: 0.02758 },
  { name: 'M', value: 0.02406 },
  { name: 'W', value: 0.0236 },
  { name: 'F', value: 0.02288 },
  { name: 'G', value: 0.02015 },
  { name: 'Y', value: 0.01974 },
  { name: 'P', value: 0.01929 },
  { name: 'B', value: 0.01492 },
  { name: 'V', value: 0.00978 },
  { name: 'K', value: 0.00772 },
  { name: 'J', value: 0.00153 },
  { name: 'X', value: 0.0015 },
  { name: 'Q', value: 0.00095 },
  { name: 'Z', value: 0.00074 },
];

export const dates = [
  { name: '2018-01-27T05:00:00.000Z', value: 56 },
  { name: '2018-01-28T05:00:00.000Z', value: 38 },
  { name: '2018-01-29T05:00:00.000Z', value: 61 },
  { name: '2018-01-30T05:00:00.000Z', value: 77 },
  { name: '2018-01-31T05:00:00.000Z', value: 19 },
  { name: '2018-02-01T05:00:00.000Z', value: 70 },
  { name: '2018-02-02T05:00:00.000Z', value: 93 },
  { name: '2018-02-03T05:00:00.000Z', value: 80 },
  { name: '2018-02-04T05:00:00.000Z', value: 99 },
  { name: '2018-02-05T05:00:00.000Z', value: 31 },
  { name: '2018-02-06T05:00:00.000Z', value: 23 },
  { name: '2018-02-07T05:00:00.000Z', value: 13 },
  { name: '2018-02-08T05:00:00.000Z', value: 81 },
];
