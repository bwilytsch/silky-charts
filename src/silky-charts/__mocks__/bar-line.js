/* ignore file coverage */

export const data = [
  { name: 'foo', apples: 18, bananas: 16, cherries: 17, dates: 1 },
  { name: 'bar', apples: 14, bananas: 14, cherries: 19, dates: 15 },
  { name: 'baz', apples: 17, bananas: 13, cherries: 18, dates: 9 },
  { name: 'qux', apples: 14, bananas: 13, cherries: 2, dates: 13 },
];
export const dataWidthDates = [
  {
    name: '2018-12-30T05:00:00.000Z',
    apples: 18,
    bananas: 16,
    cherries: 17,
    dates: 1,
  },
  {
    name: '2018-12-31T05:00:00.000Z',
    apples: 14,
    bananas: 14,
    cherries: 19,
    dates: 15,
  },
  {
    name: '2019-01-01T05:00:00.000Z',
    apples: 17,
    bananas: 13,
    cherries: 18,
    dates: 9,
  },
  {
    name: '2019-01-02T05:00:00.000Z',
    apples: 14,
    bananas: 13,
    cherries: 2,
    dates: 13,
  },
];
