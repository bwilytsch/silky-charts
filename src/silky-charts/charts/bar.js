// @flow
import type { Bar } from '../types'
import { select, selectAll, selection } from 'd3-selection' // eslint-disable-line
import { axisBottom, axisLeft } from 'd3-axis'
import { areAllDate } from '../utils'
import {
  drawGrid,
  drawLabels,
  drawReferenceLine,
  getXScale,
  getYScale,
  highValues,
} from './helpers'
import 'd3-transition'

export default ({
  currency,
  currencyshortNotation,
  data,
  grid,
  horizontal,
  localization,
  margin = { top: 60, right: 60, bottom: 60, left: 60 },
  ref,
  referenceLine,
  shortNotation,
  size,
  source,
  title,
  valueOnBars,
  xAxisLabel,
  yAxisLabel,
}: Bar) => {
  const svg = select(ref)
  const height = size.height - margin.top - margin.bottom
  const width = size.width - margin.left - margin.right
  const drawLabel = drawLabels(svg, size, margin)
  // const areNamesDate = areAllDate(data.map(d => d.name))
  const areNamesDate = false
  const isValueHidden = x => height - yScale(x) < 50

  const chart = svg
    .append('g')
    .attr('transform', `translate(${margin.left}, ${margin.top})`)

  const xScale = getXScale(areNamesDate ? 'time' : 'band', data, width)
  const yScale = getYScale('linear', data, height)

  const xAxis = chart
    .append('g')
    .attr('class', 'x-axis')
    .attr('transform', `translate(0, ${height})`)
    .call(axisBottom(xScale))

  const yAxis = chart
    .append('g')
    .attr('class', 'y-axis')
    .call(axisLeft(yScale))

  // Translate and rotate the x axis labels when they are not instance of Date
  if (!areNamesDate) {
    xAxis
      .selectAll('text')
      .attr('text-anchor', 'end')
      .attr('transform', 'translate(-12, 6) rotate(-50)')
  }

  // Extend x axis path length if the names values are instance of Date
  if (areNamesDate) {
    chart.select('.x-axis path.domain').attr('d', () => `M0,0.5V0.5H${width}`)
  }

  if (grid) {
    chart
      .append('g')
      .attr('class', 'grid')
      .call(drawGrid(horizontal, xScale, height, yScale, width))
  }

  const barGroup = chart
    .append('g')
    .selectAll()
    .data(data)
    .enter()
    .append('g')

  barGroup
    .append('rect')
    .attr('class', 'bar')
    .attr('x', ({ name }) =>
      areNamesDate
        ? xScale(new Date(name)) - width / data.length / 2.65
        : xScale(name)
    )
    .attr('y', ({ value }) => yScale(value))
    .attr('height', ({ value }) => height - yScale(value))
    .attr(
      'width',
      areNamesDate ? width / (data.length * 1.25) : xScale.bandwidth()
    )

  if (valueOnBars && !highValues(data)) {
    barGroup
      .append('text')
      .attr('class', ({ value }) =>
        isValueHidden(value) ? 'value out' : 'value'
      )
      .attr('transform', ({ name, value }) => {
        const x = areNamesDate
          ? xScale(new Date(name))
          : xScale(name) + xScale.bandwidth() / 2
        const y = isValueHidden(value) ? yScale(value) - 20 : yScale(value) + 30
        return `translate(${x}, ${y})`
      })
      .attr('text-anchor', 'middle')
      .text(({ value }) => value)
  }

  barGroup
    .selectAll('.bar')
    .on('mouseenter', function({ value: currentValue }) {
      referenceLine && drawReferenceLine(chart, yScale(currentValue), width)

      if (valueOnBars) {
        const _parentNode = this.parentNode
        selectAll('.value').text(function({ value }) {
          const divergence = value - currentValue
          const text = divergence > 0 ? `+${divergence}` : `${divergence}`
          return _parentNode === this.parentNode ? '' : text
        })
      }
    })
    .on('mouseleave', function() {
      valueOnBars && selectAll('.value').text(({ value }) => value)
      referenceLine && chart.select('.reference-line').remove()
    })

  if (title) {
    drawLabel('title', title)
  }

  if (xAxisLabel) {
    drawLabel('xAxis', xAxisLabel)
  }

  if (yAxisLabel) {
    drawLabel('yAxis', yAxisLabel)
  }

  if (source) {
    drawLabel('source', source)
  }

  return chart
}
