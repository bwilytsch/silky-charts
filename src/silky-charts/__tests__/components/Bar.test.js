import React from 'react';
import { Bar } from '../../';
import { create } from 'react-test-renderer';
import { data, dataWidthDates } from '../../__mocks__/bar';
import createNodeMock from '../../__mocks__/createNodeMock';
import 'jest-styled-components';

test('render correctly', () => {
  const tree = create(<Bar data={data} />, { createNodeMock }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with width', () => {
  const tree = create(<Bar data={data} width={400} />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with height', () => {
  const tree = create(<Bar data={data} height={400} />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with width and height', () => {
  const tree = create(<Bar data={data} width={400} height={300} />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with x axis label', () => {
  const tree = create(<Bar data={data} xAxisLabel="foo" />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with y axis label', () => {
  const tree = create(<Bar data={data} yAxisLabel="foo" />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with dates as name', () => {
  const tree = create(<Bar data={dataWidthDates} />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly width horizontal grid', () => {
  const tree = create(<Bar data={data} grid />, { createNodeMock }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly width vertical grid', () => {
  const tree = create(<Bar data={data} grid horizontal />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly X axis label rotation', () => {
  const tree = create(<Bar data={data} xAxisLabelRotation />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly showing the value on bars', () => {
  const tree = create(<Bar data={data} showValue />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly showing the value and divergence on bars', () => {
  const tree = create(<Bar data={data} showValue showDivergence />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('call onMouseEnter and update divergence on bars', () => {
  const handleOnMouseEnter = jest.fn();
  const renderer = create(
    <Bar
      data={data}
      onMouseEnter={handleOnMouseEnter}
      showValue
      showDivergence
    />,
    {
      createNodeMock,
    }
  );
  const instance = renderer.root;
  const bar = instance.findByProps({ className: 'data' }).children[0];

  bar.props.onMouseEnter({ target: { getAttribute: jest.fn(x => 1) } });
  expect(handleOnMouseEnter).toHaveBeenCalled();

  const tree = renderer.toJSON();
  expect(tree).toMatchSnapshot();
});

test('call onMouseLeave', () => {
  const handleOnMouseLeave = jest.fn();
  const renderer = create(
    <Bar data={data} onMouseLeave={handleOnMouseLeave} />,
    {
      createNodeMock,
    }
  );
  const instance = renderer.root;
  const bar = instance.findByProps({ className: 'data' }).children[0];

  bar.props.onMouseLeave();
  expect(handleOnMouseLeave).toHaveBeenCalled();
});
