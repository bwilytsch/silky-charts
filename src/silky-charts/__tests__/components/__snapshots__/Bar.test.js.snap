// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should call onMouseEnter and update divergence on bars 1`] = `
.c1 {
  fill: rgb(90,200,250);
  height: 100.10752688172045px;
  width: 181.73076923076923px;
}

.c1:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c3 {
  fill: rgb(90,200,250);
  height: 368.81720430107526px;
  width: 181.73076923076923px;
}

.c3:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c4 {
  fill: rgb(90,200,250);
  height: 490px;
  width: 181.73076923076923px;
}

.c4:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c5 {
  fill: rgb(90,200,250);
  height: 421.505376344086px;
  width: 181.73076923076923px;
}

.c5:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c0 {
  height: 540px;
  width: 960px;
}

.c2 {
  fill: rgb(255,255,255);
  pointer-events: none;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-7"
  size={
    Object {
      "height": 540,
      "width": 960,
    }
  }
>
  <g
    className="container"
    transform="translate(40, 20)"
  >
    <g
      className="axis-x "
      transform="translate(0, 490)"
    />
    <g
      className="axis-y "
    />
    <g
      className="data"
    >
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 100.10752688172045,
            "width": 181.73076923076923,
          }
        }
        value="+18"
        x={34.61538461538464}
        y={389.89247311827955}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 100.10752688172045,
            "width": 181.73076923076923,
          }
        }
        textAnchor="middle"
        transform="translate(125.48076923076925, 419.89247311827955)"
      >
        +18
      </text>
      <rect
        className="bar c3"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 368.81720430107526,
            "width": 181.73076923076923,
          }
        }
        value="+69"
        x={250.96153846153848}
        y={121.18279569892474}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 368.81720430107526,
            "width": 181.73076923076923,
          }
        }
        textAnchor="middle"
        transform="translate(341.8269230769231, 151.18279569892474)"
      >
        +69
      </text>
      <rect
        className="bar c4"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 490,
            "width": 181.73076923076923,
          }
        }
        value="+92"
        x={467.3076923076923}
        y={0}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 490,
            "width": 181.73076923076923,
          }
        }
        textAnchor="middle"
        transform="translate(558.1730769230769, 30)"
      >
        +92
      </text>
      <rect
        className="bar c5"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 421.505376344086,
            "width": 181.73076923076923,
          }
        }
        value="+79"
        x={683.6538461538462}
        y={68.49462365591398}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 421.505376344086,
            "width": 181.73076923076923,
          }
        }
        textAnchor="middle"
        transform="translate(774.5192307692308, 98.49462365591398)"
      >
        +79
      </text>
    </g>
  </g>
</svg>
`;

exports[`Should render correctly 1`] = `
.c1 {
  fill: rgb(90,200,250);
  height: 0px;
  width: 0px;
}

.c1:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-0"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
    <g
      className="data"
    >
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={19}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={70}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={93}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={80}
        x={0}
        y={0}
      />
    </g>
  </g>
</svg>
`;

exports[`Should render correctly X axis label rotation 1`] = `
.c1 {
  fill: rgb(90,200,250);
  height: 0px;
  width: 0px;
}

.c1:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-2"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
    <g
      className="data"
    >
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={19}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={70}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={93}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={80}
        x={0}
        y={0}
      />
    </g>
  </g>
</svg>
`;

exports[`Should render correctly showing the value and divergence on bars 1`] = `
.c1 {
  fill: rgb(90,200,250);
  height: 0px;
  width: 0px;
}

.c1:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c0 {
  height: 90px;
  width: 100px;
}

.c2 {
  fill: rgb(33,33,33);
  pointer-events: none;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-3"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
    <g
      className="data"
    >
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={19}
        x={0}
        y={0}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        textAnchor="middle"
        transform="translate(0, -18)"
      >
        19
      </text>
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={70}
        x={0}
        y={0}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        textAnchor="middle"
        transform="translate(0, -18)"
      >
        70
      </text>
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={93}
        x={0}
        y={0}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        textAnchor="middle"
        transform="translate(0, -18)"
      >
        93
      </text>
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={80}
        x={0}
        y={0}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        textAnchor="middle"
        transform="translate(0, -18)"
      >
        80
      </text>
    </g>
  </g>
</svg>
`;

exports[`Should render correctly showing the value on bars 1`] = `
.c1 {
  fill: rgb(90,200,250);
  height: 100.10752688172045px;
  width: 181.73076923076923px;
}

.c1:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c3 {
  fill: rgb(90,200,250);
  height: 368.81720430107526px;
  width: 181.73076923076923px;
}

.c3:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c4 {
  fill: rgb(90,200,250);
  height: 490px;
  width: 181.73076923076923px;
}

.c4:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c5 {
  fill: rgb(90,200,250);
  height: 421.505376344086px;
  width: 181.73076923076923px;
}

.c5:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c0 {
  height: 540px;
  width: 960px;
}

.c2 {
  fill: rgb(255,255,255);
  pointer-events: none;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-5"
  size={
    Object {
      "height": 540,
      "width": 960,
    }
  }
>
  <g
    className="container"
    transform="translate(40, 20)"
  >
    <g
      className="axis-x "
      transform="translate(0, 490)"
    />
    <g
      className="axis-y "
    />
    <g
      className="data"
    >
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 100.10752688172045,
            "width": 181.73076923076923,
          }
        }
        value={19}
        x={34.61538461538464}
        y={389.89247311827955}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 100.10752688172045,
            "width": 181.73076923076923,
          }
        }
        textAnchor="middle"
        transform="translate(125.48076923076925, 419.89247311827955)"
      >
        19
      </text>
      <rect
        className="bar c3"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 368.81720430107526,
            "width": 181.73076923076923,
          }
        }
        value={70}
        x={250.96153846153848}
        y={121.18279569892474}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 368.81720430107526,
            "width": 181.73076923076923,
          }
        }
        textAnchor="middle"
        transform="translate(341.8269230769231, 151.18279569892474)"
      >
        70
      </text>
      <rect
        className="bar c4"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 490,
            "width": 181.73076923076923,
          }
        }
        value={93}
        x={467.3076923076923}
        y={0}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 490,
            "width": 181.73076923076923,
          }
        }
        textAnchor="middle"
        transform="translate(558.1730769230769, 30)"
      >
        93
      </text>
      <rect
        className="bar c5"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 421.505376344086,
            "width": 181.73076923076923,
          }
        }
        value={80}
        x={683.6538461538462}
        y={68.49462365591398}
      />
      <text
        className="value c2"
        size={
          Object {
            "height": 421.505376344086,
            "width": 181.73076923076923,
          }
        }
        textAnchor="middle"
        transform="translate(774.5192307692308, 98.49462365591398)"
      >
        80
      </text>
    </g>
  </g>
</svg>
`;

exports[`Should render correctly width horizontal grid 1`] = `
.c1 path {
  stroke: transparent;
}

.c1 line {
  stroke: rgb(220,220,220);
}

.c2 {
  fill: rgb(90,200,250);
  height: 0px;
  width: 0px;
}

.c2:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-1"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="container"
    transform="translate(50, 40)"
  >
    <g
      className="grid c1"
    />
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
    <g
      className="data"
    >
      <rect
        className="bar c2"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={19}
        x={0}
        y={0}
      />
      <rect
        className="bar c2"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={70}
        x={0}
        y={0}
      />
      <rect
        className="bar c2"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={93}
        x={0}
        y={0}
      />
      <rect
        className="bar c2"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={80}
        x={0}
        y={0}
      />
    </g>
  </g>
</svg>
`;

exports[`Should render correctly width vertical grid 1`] = `
.c1 path {
  stroke: transparent;
}

.c1 line {
  stroke: rgb(220,220,220);
}

.c2 {
  fill: rgb(90,200,250);
  height: 100.10752688172045px;
  width: 181.73076923076923px;
}

.c2:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c3 {
  fill: rgb(90,200,250);
  height: 368.81720430107526px;
  width: 181.73076923076923px;
}

.c3:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c4 {
  fill: rgb(90,200,250);
  height: 490px;
  width: 181.73076923076923px;
}

.c4:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c5 {
  fill: rgb(90,200,250);
  height: 421.505376344086px;
  width: 181.73076923076923px;
}

.c5:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c0 {
  height: 540px;
  width: 960px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-3"
  size={
    Object {
      "height": 540,
      "width": 960,
    }
  }
>
  <g
    className="container"
    transform="translate(40, 20)"
  >
    <g
      className="grid c1"
    />
    <g
      className="axis-x "
      transform="translate(0, 490)"
    />
    <g
      className="axis-y "
    />
    <g
      className="data"
    >
      <rect
        className="bar c2"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 100.10752688172045,
            "width": 181.73076923076923,
          }
        }
        value={19}
        x={34.61538461538464}
        y={389.89247311827955}
      />
      <rect
        className="bar c3"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 368.81720430107526,
            "width": 181.73076923076923,
          }
        }
        value={70}
        x={250.96153846153848}
        y={121.18279569892474}
      />
      <rect
        className="bar c4"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 490,
            "width": 181.73076923076923,
          }
        }
        value={93}
        x={467.3076923076923}
        y={0}
      />
      <rect
        className="bar c5"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 421.505376344086,
            "width": 181.73076923076923,
          }
        }
        value={80}
        x={683.6538461538462}
        y={68.49462365591398}
      />
    </g>
  </g>
</svg>
`;

exports[`Should render correctly with dates as name 1`] = `
.c1 {
  fill: rgb(90,200,250);
  height: 100.10752688172045px;
  width: 187.92282635930843px;
}

.c1:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c2 {
  fill: rgb(90,200,250);
  height: 368.81720430107526px;
  width: 187.92282635930843px;
}

.c2:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c3 {
  fill: rgb(90,200,250);
  height: 490px;
  width: 187.92282635930843px;
}

.c3:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c4 {
  fill: rgb(90,200,250);
  height: 421.505376344086px;
  width: 187.92282635930843px;
}

.c4:hover {
  fill: hsl(198.79999999999995,94.1%,53.3%);
}

.c0 {
  height: 540px;
  width: 960px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-1"
  size={
    Object {
      "height": 540,
      "width": 960,
    }
  }
>
  <g
    className="container"
    transform="translate(40, 20)"
  >
    <g
      className="axis-x "
      transform="translate(0, 490)"
    />
    <g
      className="axis-y "
    />
    <g
      className="data"
    >
      <rect
        className="bar c1"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 100.10752688172045,
            "width": 187.92282635930843,
          }
        }
        value={19}
        x={31.25}
        y={389.89247311827955}
      />
      <rect
        className="bar c2"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 368.81720430107526,
            "width": 187.92282635930843,
          }
        }
        value={70}
        x={248.25}
        y={121.18279569892474}
      />
      <rect
        className="bar c3"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 490,
            "width": 187.92282635930843,
          }
        }
        value={93}
        x={464.25}
        y={0}
      />
      <rect
        className="bar c4"
        color="rgb(90, 200, 250)"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size={
          Object {
            "height": 421.505376344086,
            "width": 187.92282635930843,
          }
        }
        value={80}
        x={681.25}
        y={68.49462365591398}
      />
    </g>
  </g>
</svg>
`;
